docs/proposals/28-autonomous-shoot-clusters.md

```diff
@@ -0,0 +1,495 @@
+
+In the `gardenadm init` flow, the initial application of the `OperatingSystemConfig` by components of the `gardener-node-agent` is an example of this.
+The `gardener-node-agent` retrieves the `OperatingSystemConfig` from a `Secret`.
+Furthermore, it also checks its local `Node` resource.
+Both are not available until after the control plane starts running.
+Therefore, a fake client is useful for bootstrapping.
+
+Another example is the deployment of the extensions.
+The existing implementation requires a garden cluster with `Controller{Registration,Deployment}` resources to apply `ControllerInstallation`s.
+However, this also works with the fake client so that extensions can be deployed during bootstrapping.
+
+###### Host Network vs. Pod Network
+
+Typically, the `gardener-resource-manager` is always deployed in the pod network.
+The same is true for the Gardener extensions.
+
+In general, it is desirable to be able to use the pod network as it offers a better separation and less potential for clashes, e.g., with regards to port reuse.
+Components most likely need to be adapted to make some ports configurable so that they do not clash, e.g., metrics and webhooks ports of extensions.
+During bootstrapping, the circular dependencies between the components makes it impossible to start with the pod network.
+
+It is obvious, that the networking extension, which should create the manifests to be used to create a working pod network, can initially not run in the pod network itself.
+As `gardener-resource-manager` is utilized to deploy the manifests via a `ManagedResource`, it cannot run in the pod network initially either.
+
+Once the pod network is established, theses components can be restarted in the pod network.
+
+###### Dealing with Webhooks for Control Plane Components
+
+The minimal standard Kubernetes control plane components (`etcd`, `kube-apiserver`, `kube-controller-manager`, `kube-scheduler`) run as static pods in this setup.
+However, some extensions might want to mutate them [via webhooks](../extensions/controlplane-webhooks.md).
+Previously, this was done by mutating the `Deployment`/`StatefulSet` of the components.
+With static pods, those resources do not exist, though.
+
+To make it possible to use webhooks as before, we propose to create dummy `Deployment`/`StatefulSet` resources for the control plane components with `replicas=0`.
+These resources will be watched and updates will be translated to the manifests of the static pods.
+The manifests are then rolled out via the `OperatingSystemConfig` applied by `gardener-node-agent`.
+
+###### `etcd` Management via [`etcd-druid`](https://github.com/gardener/etcd-druid)
+
+Gardener itself will only be capable of running a minimal deployment of `etcd` for the bootstrapping process (i.e., no [`backup-restore`](https://github.com/gardener/etcd-backup-restore) sidecar or any sophisticated configuration for running `etcd` productively).
```

Comment:
Do you envision to have some integration with the BackupBucket/BackupEntry resources?

Reply:
We want to offer backup for `etcd`. Hence, we would want to have integration with `BackupBucket`/`BackupEntry` as far as I can tell.
Do you see issues in this regard?